src/client/components/pages/Courses/OfferedModal.tsx

```diff
@@ -0,0 +1,228 @@
+    term: TERM,
+    calendarYear: string,
+  };
+  /**
+   * Full details of the instance for which the offered value is being edited
+   */
+  currentCourseInstance: CourseInstanceResponseDTO
+  /**
+   * A function that will close the modal when called
+   */
+  onClose: () => void;
+  /**
+   * Handler to be invoked when the modal is saved
+   */
+  onSave: (updatedInstance: CourseInstanceUpdateDTO) => void;
+}
+
+const OfferedModal: FunctionComponent<OfferedModalProps> = ({
+  isVisible,
+  currentSemester,
+  currentCourseInstance,
+  onClose,
+  onSave,
+}): ReactElement => {
+  /**
+   * Ref to attach to the internal modal header
+   */
+  const modalHeaderRef = useRef<HTMLHeadingElement>(null);
+
+  /**
+  * Shift the focus to the modal header when it appears on the page
+  */
+  const setOfferedModalFocus = (): void => {
+    setTimeout(() => modalHeaderRef.current?.focus());
+  };
+
+  /**
+   * Store any error messages generated inside the modal
+   */
+  const [errorMessage, setErrorMessage] = useState('');
+
+  /**
+   * Track whether data is being pushed to the server
+   */
+  const [saving, setSaving] = useState(false);
+
+  type FormField = HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement;
+
+  const [form, setFormFields] = useState({
+    offered: '',
+  });
+
+  const updateFormFields = (event: ChangeEvent): void => {
+    const target = event.target as FormField;
+    setFormFields({
+      ...form,
+      [target.name]: target.value,
+    });
+  };
+
+  const instanceIdentifier = currentCourseInstance && currentSemester
+    ? `${
+      currentCourseInstance.catalogNumber
+    }, ${
+      currentSemester.term
+    } ${
+      currentSemester.calendarYear
+    }`
+    : '';
+
+  useEffect(() => {
+    if (currentSemester && currentCourseInstance) {
+      const currentTermKey = currentSemester.term.toLowerCase() as TermKey;
+      setFormFields({
+        offered: currentCourseInstance[currentTermKey].offered,
+      });
+    }
+  }, [currentSemester, currentCourseInstance]);
+
+  useEffect(() => {
+    if (isVisible) {
+      setOfferedModalFocus();
+      setErrorMessage('');
+    }
+  }, [isVisible]);
+
+  const saveOffered = async () => {
+    setSaving(true);
+    let results: CourseInstanceUpdateDTO;
+    try {
+      const currentTermKey = currentSemester.term.toLowerCase() as TermKey;
+      results = await updateCourseInstance(
+        currentCourseInstance[currentTermKey].id,
+        {
+          offered: form.offered as OFFERED,
+          preEnrollment: currentCourseInstance[currentTermKey].preEnrollment,
+          studyCardEnrollment: currentCourseInstance[currentTermKey]
+            .studyCardEnrollment,
+          actualEnrollment: currentCourseInstance[currentTermKey]
+            .actualEnrollment,
+        }
+      );
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        const serverError = error.response.data as Error;
+        setErrorMessage(serverError.message);
```

Comment:
It looks like in the case of a validation error, the value of `serverError.message` is an array of Objects, each corresponding to a different validation issue. Since we're dropping this straight in to the `ModalMessage` below, it's resulting in a White Screen of Death since arrays/objects can't be used directly as children. So we probably need some additional parsing here to handle validation errors specifically.

(This is set up the same in our other modals, so this has probably been a sleeping bug for a while that's only popping up now -- I wouldn't worry about updating those other instances in this PR, but we probably want to open a ticket for that...)